source/Calamari/Kubernetes/ResourceStatus/ResourceStatusReportWrapper.cs

```diff
@@ -96,6 +96,15 @@ public bool IsEnabled(ScriptSyntax syntax)
                 return result;
             }
 
+            
+            var kubeConfig = Path.Combine(workingDirectory, "kubectl-octo.yml");
+
+            if (environmentVars == null)
```

Comment:
Is tjhis able to be null? Nothing else seems to do that check here. Ideally we should have null checking on for this project but it should be probably done earlier in the flow rather than mid way through.

Reply:
Yes, we found it to be null when we ran it. That's why we added this checking.
In the other wrapper the same has been done as well: https://github.com/OctopusDeploy/Calamari/blob/b7c5f152fb2a768761ef45372dcd79f554823a66/source/Calamari/Kubernetes/KubernetesContextScriptWrapper.cs#L60-L63

I think the root cause is this default value here: https://github.com/OctopusDeploy/Calamari/blob/b7c5f152fb2a768761ef45372dcd79f554823a66/source/Calamari.Common/Features/Scripting/ScriptEngine.cs#L45

A proper fix should remove this default value, and add an empty Dictionary initialisation everywhere this method is invoked.

Reply:
> A proper fix should remove this default value, and add an empty Dictionary initialisation everywhere this method is invoked.

Sounds good :D